### ISSUE TITLE
Avoid Committing Internal Package Credentials

### ISSUE THREAD

**Issue by CDTR-MattConroy** [GlitchEnzo/NuGetForUnity#670]
## Description

I have an internal package feed that requires configuration in the NuGet.config. However because NuGetForUnity uses it's own NuGet.config, I'm unable to add the package feed without either committing it to the repository or ignoring the entire file. Is there a way it can use the system level NuGet.config, which is already configured for our internal package feeds for other applications, or some other way that I can avoid sharing credentials?

**Comment by CDTR-MattConroy** [comment]
Hey @JoC0de, I appreciate you're likely busy, but is there any workaround for this in lieu of a fix?

**Comment by CDTR-MattConroy** [comment]
I'm having bizarre issues with this with regards to casing it seems.
On my local machine I have my internal package feed configured named "GitHub", and this works fine. Nuget, NuGetForUnity and NuGetForUnity.Cli all pick this up correctly.
On my build agents, I have the same feed (different credentials), configured named "github", and this fails with an Unauthorized error, shown below with the organisation replaced with `<org>`:

```
Initialization of api client for 'https://nuget.pkg.github.com/<org>/index.json' failed so we can't use it. error (cached): System.Net.Http.HttpRequestException: The request to 'https://nuget.pkg.github.com/<org>/index.json' failed with status code 'Unauthorized' and message:
   at NugetForUnity.PackageSource.NugetApiClientV3.EnsureResponseIsSuccessAsync(HttpResponseMessage response) in /_/src/NuGetForUnity/Editor/PackageSource/NugetApiClientV3.cs:line 458
   at NugetForUnity.PackageSource.NugetApiClientV3.GetStringFromServerAsync(NugetPackageSourceV3 packageSource, String url, CancellationToken cancellationToken) in /_/src/NuGetForUnity/Editor/PackageSource/NugetApiClientV3.cs:line 763
   at NugetForUnity.PackageSource.NugetApiClientV3.InitializeApiAddressesAsync(NugetPackageSourceV3 packageSource) in /_/src/NuGetForUnity/Editor/PackageSource/NugetApiClientV3.cs:line 633
   at NugetForUnity.PackageSource.NugetApiClientV3.EnsureInitializedAsync(NugetPackageSourceV3 packageSource) in /_/src/NuGetForUnity/Editor/PackageSource/NugetApiClientV3.cs:line 596.
```

I've tested the same credentials locally and it produces the same result. NuGet is fine with either, only NuGetForUnity + NuGetForUnity.Cli seem to be affected. Any thoughts?

Let me know if you want me to open this as a new issue!

**Comment by JoC0de** [comment]
I can see it is not case insensitive:
https://github.com/JoC0de/NuGetForUnity/blob/c8205ffcd945cd176f4493a8ab25db093efef779/src/NuGetForUnity/Editor/Configuration/NugetConfigFile.cs#L248

**Comment by JoC0de** [comment]
I fixed it in #700 it is in the latest release.
